src/model.ts

```diff
@@ -76,10 +96,23 @@ export abstract class Model<T extends Model<T>> {
     return pk.query()
   }
 
+  secondaries(keys?: Exact[]) {
+    const labels: { [label in Labels]?: string } = {}
+    const secondaryKeys = keys || this.keys()
+    for (const key of secondaryKeys) {
+      if (isSecondary(key.index)) {
+        labels[key.index.options.label] = key.query()
+      }
+    }
+
+
+    return labels
+  }
+
   /**
    *
-   * Will save the model to the database. If the primary key has changed,
-   * the previous entry in the database will be deleted first.
+   * Will save the model to the database, as well as any shadows. If the primary key has changed,
+   * the previous entry in the database will be deleted first, this also runs for primary shadow keys.
    *
    */
   async save() {
```

Comment:
If I am not mistaken, this function is also a bit buggy (it is not a newly introduced bug, this bug is present in the current published version as well): The snapshot is NOT updated, so if you change the model in a way that results in change of primary:

- upon first save, the previous instance is deleted.
- upon second save, the previous instance is again deleted (while it probably doesn't exist).

The same issue is also present for shadow primaries. Perhaps all primaries should be updated upon saving?

Reply:
> is there a reason to not update the snapshot when the primary key updates?

Yes. The snapshot needs to be updated after the model is _saved to the database_, which happens way after the primary key updates. When saving to db, you definitely need the snapshot in order to clean up the previous object (if need be), which means the information must be preserved through updates to the primary key before the instance is saved.

```ts
const bob = await indexBy(Name).exact('Bob').get(Person)

bob.name = 'Rob' // 👉 primary key is updated
bob.name = 'Jacob' // 👉 primary key is updated

bob.save()
// ☝️Here we need to remember the original name, 'Bob',
//   as we need to delete the record with name 'Bob' from db.
//   This means we had to preserve the "snapshot" (the original name)
//   despite having updated the primary key.
```

Reply:
Sure, I understand the purpose of the snapshot. Is this happening if the save function is called twice on the same instance? 
```
const bob = await indexBy(Name).exact('Bob').get(Person)

bob.name = 'Rob' // 👉 primary key is updated
bob.name = 'Jacob' // 👉 primary key is updated

bob.save()

bob.name = "John"

bob.save()
```

Reply:
Sure, I was thinking that the snapshot could be updated in the comparison after deleting the existing entry by primary, so that the subsequent save call would have the correct snapshot
```
if (this.__snapshot && primary !== this.__snapshot) {
      await data.remove(this.__snapshot)
      this.__snapshot = primary
}
```

Reply:
Exactly. And I suspect the same should go for primary snapshots (they need to get updated after all deletions and updates are conducted to the db).

Reply:
okay, cool. That's an easy enough fix